dev/Telemetry/RuntimeProfiler.cpp

```diff
@@ -81,28 +148,24 @@ namespace RuntimeProfiler {
                 return;
             }
 
-            UINT32      ArraySize = (UINT32)(m_Counts.max_size());
-            bool        bOverflow = ((UINT32)(m_cMethods) >= ArraySize);
-            UINT32      cMethods = (UINT32)m_cMethods;
-            bool        bStringOverflow = false;
-            UINT16      cMethodsLogged = 0;
-            bool        bSeparator = false;
-            
             //  Each entry will look like this:
             //  [XX|YYYY]:ZZZ
             //  Conservatively accounting for 20 characters per entry depending
             //  on the length of the numbers.
             WCHAR       OutputBuffer[20 * TableSize];
-            size_t      cchDest = ARRAYSIZE(OutputBuffer);
-   
-            // min
-            cMethods = std::min(cMethods, ArraySize);
-   
-            PWSTR       pszDest = &(OutputBuffer[0]);
-            
-            for (UINT32 ii = 0; ii < cMethods; ii++)
+            WCHAR       Number[15];     //  MAX_INT is 4294967295
+            UINT32      cMethods = (UINT32)m_cMethods;
```

Comment:
[Use static_cast](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es49-if-you-must-use-a-cast-use-a-named-cast)